test/oidc-utils.spec.ts

```diff
@@ -55,36 +20,57 @@ describe('OidcUtils', (): void => {
+        });
+
         it('should export step outputs when CLI succeeds', async (): Promise<void> => {
-            const mockOutput: ExecOutput = {
-                exitCode: 0,
-                stdout: '{"AccessToken":"abc","Username":"tester"}',
-                stderr: '',
-            };
-            (getExecOutput as jest.Mock).mockResolvedValueOnce(mockOutput);
+            const mockOutput: string = 'AccessToken: abc Username: tester';
+
+            const mockRunCli: any = jest.spyOn(Utils, 'runCliAndGetOutput').mockResolvedValueOnce(mockOutput);
 
             const result: string | undefined = await OidcUtils.exchangeOIDCTokenAndExportStepOutputs(creds);
 
             expect(result).toBe('abc');
             expect(core.setOutput).toHaveBeenCalledWith('oidc-token', 'abc');
             expect(core.setOutput).toHaveBeenCalledWith('oidc-user', 'tester');
+
+            mockRunCli.mockRestore();
         });
 
-        it('should throw if CLI command fails', async (): Promise<void> => {
-            const mockOutput: ExecOutput = {
-                exitCode: 1,
-                stdout: '',
-                stderr: 'boom',
-            };
-            (getExecOutput as jest.Mock).mockResolvedValueOnce(mockOutput);
+        it('should correctly set step outputs for CLI token exchange', async (): Promise<void> => {
+            const mockOutput: string = 'AccessToken: cli-token Username: cli-user';
 
-            await expect(OidcUtils.exchangeOIDCTokenAndExportStepOutputs(creds)).rejects.toThrow('CLI command failed with exit code 1: boom');
-        });
+            const mockRunCli: any = jest.spyOn(Utils, 'runCliAndGetOutput').mockResolvedValueOnce(mockOutput);
 
-        it('should throw if CLI execution throws', async (): Promise<void> => {
-            (getExecOutput as jest.Mock).mockRejectedValueOnce(new Error('exec failed'));
+            const result: string | undefined = await OidcUtils.exchangeOIDCTokenAndExportStepOutputs(creds);
+
+            expect(result).toBe('cli-token');
+            expect(core.setOutput).toHaveBeenCalledWith('oidc-token', 'cli-token');
+            expect(core.setOutput).toHaveBeenCalledWith('oidc-user', 'cli-user');
+
+            mockRunCli.mockRestore();
+        });
 
-            await expect(OidcUtils.exchangeOIDCTokenAndExportStepOutputs(creds)).rejects.toThrow('Failed to exchange OIDC token: exec failed');
+        it('should correctly set step outputs for manual token exchange', async (): Promise<void> => {
+            // Arrange
+            const dummyToken: any = [
+                Buffer.from(JSON.stringify({ alg: 'none' })).toString('base64'), // Header
+                Buffer.from(JSON.stringify({ sub: 'jfrt@dummy-user' })).toString('base64'), // Payload
+                '', // No signature for testing
+            ].join('.');
+            const mockUsername: string = 'manual-user';
+
+            jest.spyOn(OidcUtils, 'extractTokenUser').mockReturnValueOnce(mockUsername);
```

Comment:
Is this spy really needed? Won't `extractTokenUser` just extract the username from the token above?